### PR TITLE
F#8

### PR DESCRIPTION
**Pull request by vzarytovskii**
We want version 8 to be default in rc1, we'll still get nullables into 8, and likely records interop.

### CODE REVIEW

**Comment by abonie** [comment]
I am wondering, shouldn't these unit tests stick to `withLangVersionPreview`? I would assume these tests should also work with preview going forward, if we change the tests we might miss some regression, no? And if there is an explicit decision that something should work differently under preview, only then would we change the test from `withLangVersionPreview` to `withLangVersion80`

**Review by abonie — APPROVED** [review]

**Review by 0101 — APPROVED** [review]

File: src/Compiler/Facilities/LanguageFeatures.fs [hunk 1]
```diff
@@ -152,31 +154,33 @@ type LanguageVersion(versionText) =
...
+                LanguageFeature.DiagnosticForObjInference, languageVersion80
+                LanguageFeature.WarningWhenTailRecAttributeButNonTailRecUsage, languageVersion80
+                LanguageFeature.StaticLetInRecordsDusEmptyTypes, languageVersion80
+                LanguageFeature.StrictIndentation, languageVersion80
+                LanguageFeature.ConstraintIntersectionOnFlexibleTypes, languageVersion80
+                LanguageFeature.UnmanagedConstraintCsharpInterop, languageVersion80
+                LanguageFeature.WhileBang, languageVersion80
+
                 // F# preview
                 LanguageFeature.FromEndSlicing, previewVersion
```
Comment (psfinaki) [inline src/Compiler/Facilities/LanguageFeatures.fs:183]: What about this one?

**Review by psfinaki — APPROVED** [review]

**Comment by KevinRansom** [comment]
No, the tests need updating to the published language version

**Comment by vzarytovskii** [comment]
@T-Gro with langversion 8 by default, some tests are failing with:
```
failwith "getTypeRefsInType: modified"
```
Like `innerpolly` test, if you change it to run something like
```
[<Fact>]
let ``innerpoly-FSC_DEBUG`` () = 
    singleTestBuildAndRunAux "core/innerpoly" [ "--multiemit-" ] FSC_DEBUG
```

and run it
```
dotnet test .\tests\fsharp.Compiler.ComponentTests\FSharp.Compiler.ComponentTests.fsproj -c Release /p:BUILDING_USING_DOTNET=false --filter Miscellaneous.FsharpSuiteMigrated_CoreTests.innerpoly --no-restore -f net8.0
```

Without running multiemit, the fail is a bit mor cryptic:
```
 All errors:
[{ Error = Error 193
   Range = { StartLine = 1
             StartColumn = 0
             EndLine = 1
             EndColumn = 0 }
   NativeRange = (1,0--1,0)
   Message =
    "internal error: Unable to load one or more of the requested types.
Method 'Specialize' on type 'func@310-9' from assembly 'FSI-ASSEMBLY, Version=0.0.0.32767, Culture=neutral, PublicKeyToken=null' tried to implicitly override a method with weaker type parameter constraints."
   SubCategory = "interactive" }]
```

I will probably move unmanaged feature back to preview, until we fix it.

**Comment by vzarytovskii** [comment]
A bunch of autocomplete tests are failing due to Obsolete handling (https://dev.azure.com/dnceng-public/public/_build/results?buildId=375838&view=ms.vss-test-web.build-test-results-tab&runId=7996012&resultId=100772&paneView=debug)

@edgarfgp if I remember correctly, you were working on something related, do you think it's by design?

**Review by edgarfgp — APPROVED** [review]

**Comment by vzarytovskii** [comment]
Just a note, I had to disable couple of tests in the old vs language service, which is not in use.
For some reason tests weren't working there, but similar ones were fine in FCS.

**Pull request merged**